XYZDrivers/src/main/java/com/xyzdrivers/controllers/ClaimsController.java

```diff
@@ -57,7 +58,10 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
             String reason = request.getParameter("reason");
             String amount = request.getParameter("amount");
             
-            Claim claim = new Claim("null", date, reason, "NEW", Float.parseFloat(amount));
+            HttpSession session = request.getSession();
+            String username = (String)session.getAttribute("username");
```

Comment:
Redundant cast?

Reply:
Required, can't implicitly cast String to Object otherwise. Surprised me too.